.github/scripts/label_promoted_commits.py

```diff
@@ -44,49 +61,26 @@ def main():
     processed_prs = set()
     # Print commit information
     for commit in commits.commits:
+        print(f'Commit sha is: {commit.sha}')
         match = pr_pattern.search(commit.commit.message)
         if match:
-            search_url = f'https://api.github.com/search/issues'
-            query = f"repo:{args.repository} is:pr is:closed sha:{commit.sha}"
-            params = {
-                "q": query,
-            }
-            headers = {
-                "Authorization": f"token {github_token}",
-                "Accept": "application/vnd.github.v3+json"
-            }
-            response = requests.get(search_url, headers=headers, params=params)
-            prs = response.json().get("items", [])
-            for pr in prs:
-                refs_match = re.findall(r'Refs (?:#|https.*?)(\d+)', pr["body"])
-                ref = re.search(r'branch-(\d+\.\d+)', args.ref)
-                if refs_match and ref:
-                    pr_number = int(refs_match[0])
-                    backport_pr_number = int(match.group(1))
-                    if pr_number in processed_prs:
-                        continue
-                    label_to_add = f'backport/{ref.group(1)}-done'
-                    label_to_remove = f'backport/{ref.group(1)}'
-                    pr = repo.get_pull(pr_number)
-                    current_labels = pr.get_labels()
-                    current_label_names = [label.name for label in current_labels]
-                    if label_to_remove in current_label_names:
-                        pr.remove_from_labels(label_to_remove)
-                    if label_to_add not in current_label_names:
-                        pr.add_to_labels(label_to_add)
-                    comment = f'Closed via {commit.sha}'
-                    add_comment_and_close_pr(backport_pr_number, repo, comment)
-                    processed_prs.add(pr_number)
-                else:
-                    assert re.match(r'/master', args.ref)
-                    pr_number = pr["number"]
-                    pr = repo.get_pull(pr_number)
-                    label_to_add = args.label
-                    current_labels = pr.get_labels()
-                    current_label_names = [label.name for label in current_labels]
-                    if label_to_add not in current_label_names:
-                        pr.add_to_labels(label_to_add)
-                processed_prs.add(pr_number)
+            pr_number = int(match.group(1))
+            pr = repo.get_pull(pr_number)
+            refs_pr = re.findall(r'Refs (?:#|https.*?)(\d+)', pr.body)
+            target_branch = re.search(r'branch-(\d+\.\d+)', args.ref)
+            if refs_pr and target_branch:
```

Comment:
i'd suggest put:

```py
def mark_backport_done(parent_pr, branch, sha):
    label_to_remove = f'backport/{branch}'
    label_to_add = f'{label_to_remove}-done'
    current_labels = [label.name for label in pr.get_labels()]
    if label_to_remove in current_labels:
        parent_pr.remove_from_labels(label_to_remove)
    if label_to_add not in current_labels:
        parent_pr.add_to_labels(label_to_add)

def main():
    target_branch = re.search(r'branch-(\d+\.\d+)', args.ref):
    # ...
    processed_prs = set()
    # ...
    for commit in commits.commits:
        # ...
        if pr_number in processed_prs:
             continue
        if target_branch:
            # release branch
            parent_pr = repo.get_pull(pr_number)
            refs_pr = re.findall(r'Refs (?:#|https.*?)(\d+)', pr.body)
            if refs_pr:
            # 1. change the backport label of the parent PR to note that
            #    we've merge the corresponding backport PR
            # 2. close the backport PR and leave a comment on it to note 
            #    that it has been merged with a certain git commit,
            mark_backport_done(parent_pr, commit)
            backport_pr_number = int(refs_pr[0])
            add_comment_and_close_pr(backport_pr_number, repository, comment)
        else:
            # master branch
```

for a couple reasons:

1. don't query parent PR multiple times
2. give the variables representing parent PR and backport PR better names
3. flatten the calling chain, for better readability
4. do not handle the case where we don't have `Ref` in the `else` branch even if the PR's target branch is *not* "master"
5. push the label naming logic down to `mark_backport_done()`, so it's more obvious what it is supposed to do.